pkgs/tools/package-management/nix/common.nix

```diff
@@ -174,6 +189,8 @@ self = stdenv.mkDerivation {
     (darwinMinVersionHook "10.13")
   ];
 
+  NIX_LDFLAGS = lib.optionalString (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isStatic)
+      "-framework CoreFoundation -framework SystemConfiguration";
```

Comment:
Other packages use `buildInputs` for this, and actual `CoreFoundation` (etc.) packages.

Reply:
I don't think adding the package will actually cause the build to link against the framework. The issue I am adding these is that I see missing symbols when linking libcurl.

Reply:
Generally these forced linkages mean something is broken with the build upstream, I think.

(Also the separate framework packages are deprecated compatibility stubs now and do nothing at all.)